bmc-acf/acf_manager.cpp

```diff
@@ -0,0 +1,526 @@
+        accessControlFile.data(), accessControlFile.size(),
+        publicKeyFile.data(), publicKeyFile.size(), sDecodedAsn);
+
+    if (CeLogin::CeLoginRc::Success == sRc)
+    {
+        log<level::INFO>("Signature verified");
+
+        std::string mSerialNumber({});
+        mSerialNumber = readMachineSerialNumberProperty(
+            DBUS_INVENTORY_SYSTEM_OBJECT, DBUS_INVENTORY_ASSET_INTERFACE,
+            DBUS_SERIAL_NUM_PROP);
+
+        // If serial number is empty on machine set as UNSET for check with acf
+        if (mSerialNumber.empty())
+        {
+            mSerialNumber = UNSET_SERIAL_NUM_KEYWORD;
+        }
+        sJsonData = (CeLogin::CeLoginJsonData*)OPENSSL_malloc(
+            sizeof(CeLogin::CeLoginJsonData));
+        if (sJsonData)
+        {
+            memset(sJsonData, 0x00, sizeof(CeLogin::CeLoginJsonData));
+        }
+        else
+        {
+            log<level::ERR>("malloc failed");
+            elog<InternalFailure>();
+        }
+
+        if (CeLogin::CeLoginRc::Success == sRc)
+        {
+            // Verify system serial number is in machine list (and get the
+            // authorization)
+            const uint64_t serialNumberLengthParm = mSerialNumber.length();
+            sRc = CeLogin::decodeJson(
+                (const char*)sDecodedAsn->sourceFileData->data,
+                sDecodedAsn->sourceFileData->length, mSerialNumber.c_str(),
+                serialNumberLengthParm, *sJsonData);
+            if (CeLogin::CeLoginRc::Success == sRc)
+            {
+                log<level::INFO>("Serial Number matches");
+            }
+            else
+            {
+                log<level::ERR>("Serial Number does not match");
+            }
+        }
+    }
+    else if (CeLogin::CeLoginRc::SignatureNotValid == sRc)
+    {
+        log<level::ERR>("Signature is not valid");
+    }
+    else
+    {
+        log<level::ERR>("Error: ", entry("sRc=%d", (int)sRc));
+    }
+    // Check if time is expired
+    if (CeLogin::CeLoginRc::Success == sRc)
+    {
+        uint64_t sExpirationTime = 0;
+        std::time_t timeSinceUnixEpocInSecondsParm = std::time(NULL);
+        sRc = CeLogin::isTimeExpired(sJsonData, sExpirationTime,
+                                     timeSinceUnixEpocInSecondsParm);
+        if (CeLogin::CeLoginRc::Success != sRc)
+        {
+            log<level::ERR>("ACF time expired");
+        }
+    }
+    if (CeLogin::CeLoginRc::Success == sRc)
+    {
+        // YYYY-MM-DD
+        char sTimeStr[20];
+        sprintf(sTimeStr, "%04u-%02u-%02u", sJsonData->mExpirationDate.mYear,
+                sJsonData->mExpirationDate.mMonth,
+                sJsonData->mExpirationDate.mDay);
+        sDate = sTimeStr;
+    }
+
+    if (sDecodedAsn)
+    {
+        CELoginSequenceV1_free(sDecodedAsn);
+    }
+    if (sJsonData)
+    {
+        OPENSSL_free(sJsonData);
+    }
+    return sRc;
+}
+
+acf_info ACFCertMgr::installACF(std::vector<uint8_t> accessControlFile)
```

Comment:
you need to modify your interface yaml file to specify it can throw InternalFailure and InvalidCertificate